### PR TITLE
added integration tests for sources

### LINKED ISSUE #72: integration test for supported providers
### Please describe your feature request:
Currently, there is no integration test for supported search engines/providers that can be implemented.

- [x] Shodan
- [x] Censys
- [x] Fofa
- [x] Quake
- [x] ZoomEye

Integration test will be using ENV variable to read API information.

### PR DESCRIPTION
**Pull request by LuitelSamikshya**
closes #72 
Added integration tests on providers that runs once a week.

### CODE REVIEW

File: integration-tests/source-test.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+type censysTestcases struct{}
+
+func (h censysTestcases) Execute() error {
+	token := os.Getenv("CENSYS_API_KEY")
+	if token == "" {
+		return errors.New("missing censys api key")
+	}
+	censysToken := fmt.Sprintf(`censys: [%s]`, token)
+	_ =ioutil.WriteFile(ConfigFile, []byte(censysToken), 0644)
```
Comment (Mzack9999) [inline integration-tests/source-test.go:26]: `ioutil.WriteFile` => `os.WriteFile` (`ioutil` is deprecated)

**Review by Mzack9999 — CHANGES_REQUESTED** [review]
Please replace all occurrences of `ioutil.WriteFile` with `os.WriteFile`

File: integration-tests/source-test.go (hunk 1 above)
Comment (LuitelSamikshya) [reply integration-tests/source-test.go]: Sure.

**Review by Mzack9999 — APPROVED** [review]

**Review by ehsandeep — CHANGES_REQUESTED** [review]
@LuitelSamikshya I can confirm test is being passed regardless of result, for example dummy key is used for `ZOOMEYE_API_KEY` and I can see `[✓] Test "zoomeye" passed!`.



```console
export ZOOMEYE_API_KEY=this_is_test
```

```bash
$ bash run.sh
::task~> Clean up & Build binaries files
::done::
::task~> Run integration test
Running test cases for "censys" source
[✘] Test "censys" failed: missing censys api key
Running test cases for "shodan" source
[✘] Test "shodan" failed: missing shodan api key
Running test cases for "zoomeye" source
[bash -c ./uncover -zoomeye 'app:"Atlassian JIRA"']
/bin/bash -c ./uncover -zoomeye 'app:"Atlassian JIRA"' -v

  __  ______  _________ _   _____  _____
 / / / / __ \/ ___/ __ \ | / / _ \/ ___/
/ /_/ / / / / /__/ /_/ / |/ /  __/ /    
\__,_/_/ /_/\___/\____/|___/\___/_/ v0.0.8

		projectdiscovery.io

Use with caution. You are responsible for your actions
Developers assume no liability and are not responsible for any misuse or damage.
By using uncover, you also agree to the terms of the APIs used.


[✓] Test "zoomeye" passed!
Running test cases for "fofa" source
[✘] Test "fofa" failed: missing fofa api key
Running test cases for "quake" source
[✘] Test "quake" failed: missing quake api key
::done::
```

**Comment by LuitelSamikshya** [comment]
> @LuitelSamikshya I can confirm test is being passed regardless of result, for example dummy key is used for `ZOOMEYE_API_KEY` and I can see `[✓] Test "zoomeye" passed!`.
> 
> ```
> export ZOOMEYE_API_KEY=this_is_test
> ```
> 
> ```shell
> $ bash run.sh
> ::task~> Clean up & Build binaries files
> ::done::
> ::task~> Run integration test
> Running test cases for "censys" source
> [✘] Test "censys" failed: missing censys api key
> Running test cases for "shodan" source
> [✘] Test "shodan" failed: missing shodan api key
> Running test cases for "zoomeye" source
> [bash -c ./uncover -zoomeye 'app:"Atlassian JIRA"']
> /bin/bash -c ./uncover -zoomeye 'app:"Atlassian JIRA"' -v
> 
>   __  ______  _________ _   _____  _____
>  / / / / __ \/ ___/ __ \ | / / _ \/ ___/
> / /_/ / / / / /__/ /_/ / |/ /  __/ /    
> \__,_/_/ /_/\___/\____/|___/\___/_/ v0.0.8
> 
> 		projectdiscovery.io
> 
> Use with caution. You are responsible for your actions
> Developers assume no liability and are not responsible for any misuse or damage.
> By using uncover, you also agree to the terms of the APIs used.
> 
> 
> [✓] Test "zoomeye" passed!
> Running test cases for "fofa" source
> [✘] Test "fofa" failed: missing fofa api key
> Running test cases for "quake" source
> [✘] Test "quake" failed: missing quake api key
> ::done::
> ```

The test should pass only with the use of actual key now. Some of the providers endpoint are throwing 429 error, therefore added a cron job to run tests on providers once a week.

**Pull request merged**